examples/java/spring-boot/src/main/java/uk/gov/api/springboot/infrastructure/ScheduleHandler.java

```diff
@@ -0,0 +1,33 @@
+package uk.gov.api.springboot.infrastructure;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+import uk.gov.api.springboot.domain.model.Registry;
+import uk.gov.api.springboot.domain.model.repositories.ApiStorage;
+import uk.gov.api.springboot.domain.services.FetcherService;
+
+@Component
+public class ScheduleHandler {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ScheduleHandler.class);
+  private final FetcherService fetcherService;
+  private final Registry registry;
+  private final ApiStorage storage;
+
+  public ScheduleHandler(FetcherService fetcherService, Registry registry, ApiStorage storage) {
+    this.fetcherService = fetcherService;
+    this.registry = registry;
+    this.storage = storage;
+  }
+
+  @Scheduled(fixedRateString = "${apis-fetch-rate.in.milliseconds}")
+  public void fetchAndSaveApis() {
+    LOGGER.info("Fetching and saving APIS");
+
+    registry.retrieveAll().forEach(entry -> fetcherService.fetch(entry).forEach(storage::save));
```

Comment:
I would probably say this instead:

```suggestion
    registry.retrieveAll().stream()
        .map(fetcherService::fetch)
        .flatMap(Collection::stream)
        .forEach(storage::save);
```

This gives us a slightly different breakdown of work, allows us more easily to use `.parallel()`, and could allow us to more easily add logic in there to avoid re-processing the same `Api` if we wanted

Reply:
isn't this back to what we had before, where we have to wait for all the apis to be fetched before we start saving any? Or is it worth doing that for the benefits you just mentioned?

Reply:
I've added a comment - struggling to think of a way to write a test for that

Reply:
Actually I did just try writing a test for this, and realised both of these ways of doing it save the first list of APIs before fetching the next one (I'd misunderstood how ordering works in Streams). But I think what we've got at the moment is still slightly better as we don't have the unnecessary streaming then flattening bit?

Reply:
I'm happy with that if you are :+1: